samples/04-file-transfer/README.md

```diff
@@ -130,35 +180,99 @@ java -Dedc.fs.config=samples/04-file-transfer/consumer/config.properties -jar sa
+to accept this offer.
+
+The consumer now needs to initiate a contract negotiation sequence with the provider. That sequence
+looks as follows:
+
+1. Consumer sends a contract offer to the provider (__currently, this has to be equal to the provider's offer!__)
+2. Provider validates the received offer against its own offer
+3. Provider either sends an agreement or a rejection, depending on the validation result
+4. In case of successful validation, provider and consumer store the received agreement for later reference
+
+Of course, this is the simplest possible negotiation sequence. Later on, both connectors can
+also send counter offers in addition to just confirming or declining an offer.
+
+In order to trigger the negotiation, we use the endpoint previously created in the `api` extension.
+We specify the address of the provider connector as a query parameter and set our contract offer
+in the request body. The contract offer is prepared in [contractoffer.json](contractoffer.json)
+and can be used as is. In a real scenario, a potential consumer would first need to request
+a description of the provider's offers in order to get the provider's contract offer.
+
+```bash
+curl -X POST -H "Content-Type: application/json" -d @samples/04-file-transfer/contractoffer.json "http://localhost:9191/api/negotiation?connectorAddress=http://localhost:8181/api/ids/multipart"
+```
+
+In the response we'll get a UUID that we can use to get the contract agreement negotiated between
+provider and consumer.
+
+### 3. Look up the contract agreement ID
+
+After calling the endpoint for initiating a contract negotiation, we get a UUID as the response.
+This UUID is the ID of the ongoing contract negotiation between consumer and provider.
+The negotiation sequence between provider and consumer is executed asynchronously in the background
+by a state machine. Once both provider and consumer either reach the `confirmed` or the  `declined`
+state, the negotiation is finished. We can now use the UUID to check the current status of the
+negotiation using an endpoint on the consumer side. As this end point is not part of this sample's
+API but of the actual control API, we need to authenticate ourselves to use this endpoint. For this,
+we use the `X-Api-Key` header with the same value that's set in our consumer's `config.properties`.
 
 ```bash
-curl -X POST "http://localhost:9191/api/file/test-document?connectorAddress=http://localhost:8181/&destination=/path/on/yourmachine"
+curl -X GET -H 'X-Api-Key: password' "http://localhost:9191/api/control/negotiation/{UUID}"
 ```
 
-> **Please adjust the `destination` to match your local dev machine!**
+This will return a full description of the negotiation. When the state reads `1200` (= confirmed),
+this description will also contain a contract agreement. We can now use this agreement to request
+the file. So we copy and store the agreement's ID for the next step.
```

Comment:
I would add also a little example of the structure of the result, like:
```
{
  "id": "<NEGOTIATION UUID>",
  ...
  "contractAgreement": {
    "id": "<AGREEMENT_ID>",
    ...
  },
  ...
}
```

Reply:
Good idea, added a sample output.